### ISSUE TITLE
PowerPC wheel fails to compile

### ISSUE THREAD

**Issue by mlxd** [PennyLaneAI/pennylane-lightning#119]
#### Issue description

With the recent modifications to the repository, the wheel-builder for PowerPC (ppc64le) fails to compile the C++ modules.

* *Expected behavior:* ppc64le wheels are successfully compiled.

* *Actual behavior:* Wheel-builder fails to build ppc64le.

* *Reproduces how often:* Always for PPC

#### Source code and tracebacks

Sample offending step:

```
|   gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -DVERSION_INFO="0.16.0.dev0" -I/opt/_internal/cpython-3.7.10/lib/python3.7/site-packages/pybind11/include -I./include -I/opt/_internal/cpython-3.7.10/include/python3.7m -c pennylane_lightning/src/StateVector.cpp -o build/temp.linux-ppc64le-3.7/pennylane_lightning/src/StateVector.o -O3 -W -fPIC -shared -fopenmp -std=c++17 -fvisibility=hidden
|   In file included from pennylane_lightning/src/Gates.hpp:7:0,
|                    from pennylane_lightning/src/StateVector.hpp:33,
|                    from pennylane_lightning/src/StateVector.cpp:15:
|   pennylane_lightning/src/Util.hpp: In function ‘constexpr T Pennylane::Util::SQRT2()’:
|   pennylane_lightning/src/Util.hpp:90:8: error: expected ‘(’ before ‘constexpr’
|        if constexpr (std::is_same_v<T, float>) {
|           ^~~~~~~~~
|   pennylane_lightning/src/StateVector.cpp:19:46: error: expected ‘}’ at end of input
|    template class Pennylane::StateVector<double>;
|                                                 ^
|   pennylane_lightning/src/StateVector.cpp: At global scope:
|   pennylane_lightning/src/StateVector.cpp:19:46: error: expected ‘}’ at end of input
|   pennylane_lightning/src/StateVector.cpp:19:46: error: expected ‘}’ at end of input
|   error: command 'gcc' failed with exit status 1
```

#### Additional information

Any additional information, configuration or data that might be necessary
to reproduce the issue.

**Comment by mlxd** [comment]
The root cause of this seems to be the older version of GCC/G++ in the wheel-builder image for PPC, as discussed [here](https://github.com/pypa/manylinux/issues/1012). I will use the `manylinux2014` image, which is modern enough for our needs.

**Comment by mlxd** [comment]
Fixed in PR #120. Closing.
